python/cudf/cudf/core/column/column.py

```diff
@@ -2082,6 +2066,31 @@ def as_column(
                         return cudf.core.column.Decimal32Column.from_arrow(
                             data
                         )
+                except pa.ArrowTypeError as e:
+                    # Arrow throws a type error if the input is of
+                    # mixed-precision and cannot fit into the provided
+                    # decimal type properly, see:
+                    # https://github.com/apache/arrow/pull/9948
+                    raise e
```

Comment:
Maybe I'm missing something, but isn't:

```python
try:
    something()
except error as e:
    raise e
```

the same as just:

```python
something()
```

Reply:
Correct this block, I was only trying to relay `pa.ArrowTypeError` through this and try going into `except` block below for other errors, but it doesn't make sense for a decimal type to reach that block of code and we should just let the errors propagate from here itself just for decimals case.